app/src/main/java/com/google/samples/apps/sunflower/viewmodels/GalleryViewModel.kt

```diff
@@ -16,22 +16,56 @@
 
 package com.google.samples.apps.sunflower.viewmodels
 
+import androidx.compose.runtime.State
+import androidx.compose.runtime.mutableStateOf
 import androidx.lifecycle.SavedStateHandle
 import androidx.lifecycle.ViewModel
 import androidx.lifecycle.viewModelScope
+import androidx.paging.PagingData
 import androidx.paging.cachedIn
+import com.google.samples.apps.sunflower.data.UnsplashPhoto
 import com.google.samples.apps.sunflower.data.UnsplashRepository
 import dagger.hilt.android.lifecycle.HiltViewModel
+import kotlinx.coroutines.delay
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.filterNotNull
+import kotlinx.coroutines.flow.first
+import kotlinx.coroutines.launch
 import javax.inject.Inject
 
 @HiltViewModel
 class GalleryViewModel @Inject constructor(
     savedStateHandle: SavedStateHandle,
-    repository: UnsplashRepository
+    private val repository: UnsplashRepository
 ) : ViewModel() {
 
     private var queryString: String? = savedStateHandle["plantName"]
 
-    val plantPictures =
-        repository.getSearchResultStream(queryString ?: "").cachedIn(viewModelScope)
+
+    private val _plantPictures = MutableStateFlow<PagingData<UnsplashPhoto>?>(null)
+    val plantPictures: Flow<PagingData<UnsplashPhoto>> get() = _plantPictures.filterNotNull()
+
+    private val _isRefreshing = mutableStateOf(false)
+    val isRefreshing: State<Boolean> get() = _isRefreshing
+
+    init {
+        refreshData()
+    }
+
+
+    fun refreshData() {
```

Comment:
How about expose this as a suspending function instead? This way, the logic for is refreshing can be directly in the composable

Reply:
i think if we directly expose a suspend function to our composable it will be cancelled when we rotate our screen or due to any other configuration change, whats your views?? am i wrong?

Reply:
Ah, yes, thanks. We can keep this as-is